pinecone/index_connection.go

```diff
@@ -0,0 +1,358 @@
+package pinecone
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/pinecone-io/go-pinecone/internal/gen/data"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"log"
+)
+
+type IndexConnection struct {
```

Comment:
Thanks for pulling context out of here after chatting last week. 👍

Reply:
A definite improvement for sure. Thanks for talking through it!